### PR TITLE
Keycloak 3473

### PR DESCRIPTION
**Pull request by ebondu**

### CODE REVIEW

**Comment by sebastienblanc** [comment]
This PR is hard to review since it seems that it contains commits that fixes other issues.

**Comment by ebondu** [comment]
Do want me to clean the PR to skip unrelated commits ?

**Comment by sebastienblanc** [comment]
Yes and also rebase if needed. Thx.

**Comment by ebondu** [comment]
Done. To clarify, [KEYCLOAK-3492](https://issues.jboss.org/browse/KEYCLOAK-3492) depends on [KEYCLOAK-3473](https://issues.jboss.org/browse/KEYCLOAK-3473), so we should have a look to KEYCLOAK-3492 at the same time.

File: adapters/oidc/spring-security/src/main/java/org/keycloak/adapters/springsecurity/filter/KeycloakAuthenticationProcessingFilter.java [hunk 1]
```diff
@@ -109,7 +109,7 @@ public KeycloakAuthenticationProcessingFilter(AuthenticationManager authenticati
      *
      */
     public KeycloakAuthenticationProcessingFilter(AuthenticationManager authenticationManager, RequestMatcher
-                requiresAuthenticationRequestMatcher) {
```
Comment (sebastienblanc) [inline adapters/oidc/spring-security/src/main/java/org/keycloak/adapters/springsecurity/filter/KeycloakAuthenticationProcessingFilter.java]: Has anything changed in the class ? I see only formatting changes

Comment (ebondu) [inline adapters/oidc/spring-security/src/main/java/org/keycloak/adapters/springsecurity/filter/KeycloakAuthenticationProcessingFilter.java]: Nop, this is only formatting change.

Comment (stianst) [inline adapters/oidc/spring-security/src/main/java/org/keycloak/adapters/springsecurity/filter/KeycloakAuthenticationProcessingFilter.java]: Please remove

File: adapters/oidc/spring-security/src/main/java/org/keycloak/adapters/springsecurity/filter/KeycloakAuthenticationProcessingFilter.java [hunk 2]
```diff
@@ -142,7 +142,7 @@ public Authentication attemptAuthentication(HttpServletRequest request, HttpServ
         if (AuthOutcome.FAILED.equals(result)) {
             throw new KeycloakAuthenticationException("Auth outcome: " + result);
         }
-       
+
```
Comment (stianst) [inline adapters/oidc/spring-security/src/main/java/org/keycloak/adapters/springsecurity/filter/KeycloakAuthenticationProcessingFilter.java]: Remove formatting changes

File: adapters/oidc/spring-security/src/main/java/org/keycloak/adapters/springsecurity/filter/KeycloakAuthenticationProcessingFilter.java [hunk 3]
```diff
@@ -212,7 +212,7 @@ protected void successfulAuthentication(HttpServletRequest request, HttpServletR
 
     @Override
     protected void unsuccessfulAuthentication(HttpServletRequest request, HttpServletResponse response,
-            AuthenticationException failed) throws IOException, ServletException {
+                                              AuthenticationException failed) throws IOException, ServletException {
```
Comment (stianst) [inline adapters/oidc/spring-security/src/main/java/org/keycloak/adapters/springsecurity/filter/KeycloakAuthenticationProcessingFilter.java]: Remove formatting changes

**Comment by stianst** [comment]
There is no tests here and no documentation updates. I don't understand how this is used/set either.

**Comment by ebondu** [comment]
The change is simple, it add an option to chose how to handle 401:
- by the web container that will the load the correponding error page => no way to customize the response (default/actual behavior)
- by a third component (like the sping error handler) => the handler will produce the reponse. In that case, the corresponding adapter have to set the "containerErrorPage" field to "false" before to call the authenticator. It will be the case in the KeycloakAuthenticationProcessingFilter.

If will modify the PR to remove formatting changes and update the KeycloakAuthenticationProcessingFilter to illustrate the use of the case.

Relevants tests are done in the PR https://github.com/keycloak/keycloak/pull/3741 (not sure there is something to test here. Regarding the doc I think it is more relevant to update it in https://github.com/keycloak/keycloak/pull/3741.

**Comment by stianst** [comment]
@sebastienblanc should this be closed now that 3473 is resolved?

**Comment by stianst** [comment]
KEYCLOAK-3473 is already fixed so closing this PR

**Pull request closed without merging**